sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/ReshuffleTest.java

```diff
@@ -274,4 +277,26 @@ public void testReshuffleWithTimestampsStreaming() {
 
     pipeline.run();
   }
+
+  @Test
+  @Category({ValidatesRunner.class, UsesTestStream.class})
+  public void testAssignShardFn() {
+
+    PCollection<KV<String, Integer>> input =
+        pipeline
+            .apply(
+                Create.of(ARBITRARY_KVS)
+                    .withCoder(KvCoder.of(StringUtf8Coder.of(), VarIntCoder.of())))
+            .apply(Window.into(new GlobalWindows()));
```

Comment:
The hashing in Reshuffle (and the values in ARBITRARY_KVS) is deterministic right?

We don't need to worry about this flaking and ssometimes not producing a value in every bucket?

Reply:
LGTM. This appears to be deterministic as far as I can tell.

Reply:
Is forcing windowing into GlobalWindows necessary here given that it is already a Bounded PCollection?

Reply:
@iemejia I don't believe it's necessary, no. It was intended to be explicit but I'm happy to make the test adhere to whatever standard is preferred. I'll remove the global windowing transform since that's implicit with a bounded PCollection

Reply:
@egalpin Well we can let it, it is just that it looks 'repeated' for a bounded collection.